src/client/legacy/client.rs

```diff
@@ -1374,11 +1397,11 @@ impl Builder {
     /// [`h2::client::Builder::timer`]: https://docs.rs/h2/client/struct.Builder.html#method.timer
     pub fn timer<M>(&mut self, timer: M) -> &mut Self
     where
-        M: Timer + Send + Sync + 'static,
+        M: Timer + Clone + Send + Sync + 'static,
```

Comment:
Woops. Adding this would technically be breaking.

I suppose we could just wrap it in the local `Timer` struct, and pass _that_? It will mean double Arc-dyn, which is unfortunate. Maybe we can find a clever way to indicate to `hyper::rt::Timer` that we don't need to double Arc it?

Reply:
Whoops yeah you're right, forgot about this one.

For avoiding double Arc-dyn, we can probably come up with something using `Deref`  and or `TryFrom`, though I feel it's a bit hacky. (something like, if it's from local `Timer` then don't need to wrap Arc)

I was playing around with `unwrap_or_clone` just to realize it's still experimental. :sweat_smile: https://doc.rust-lang.org/std/sync/struct.Arc.html#method.unwrap_or_clone